src/LinuxTracing/UprobesUnwindingVisitorDynamicInstrumentationTest.cpp

```diff
@@ -301,4 +302,165 @@ TEST_F(UprobesUnwindingVisitorDynamicInstrumentationTest,
   }
 }
 
+TEST_F(UprobesUnwindingVisitorDynamicInstrumentationTest,
+       VisitDynamicInstrumentationPerfEventsWithTidNamespaceTranslation) {
+  constexpr pid_t kPidTargetNamespace = 42;
+  constexpr pid_t kTidTargetNamespace = 43;
+  constexpr pid_t kPidRootNamespace = 1042;
+  constexpr pid_t kTidRootNamespace = 1043;
+  visitor_.SetInitialTidToRootNamespaceTidMapping(
+      {{kPidTargetNamespace, kPidRootNamespace}, {kTidTargetNamespace, kTidRootNamespace}});
+
+  constexpr pid_t kPidUnkown = 44;
+  constexpr pid_t kTidUnkown = 45;
+
+  constexpr pid_t kTidNewTargetNamespace = 54;
+  constexpr pid_t kTidNewRootNamespace = 1054;
+
+  {
+    UserSpaceFunctionEntryPerfEvent function_entry1{
+        .timestamp = 300,
+        .data =
+            {
+                .pid = kPidTargetNamespace,
+                .tid = kTidTargetNamespace,
+                .function_id = 1,
+                .sp = 0x30,
+                .return_address = 0x02,
+            },
+    };
+
+    EXPECT_CALL(return_address_manager_, ProcessFunctionEntry(kTidRootNamespace, 0x30, 0x02))
+        .Times(1);
+    PerfEvent{function_entry1}.Accept(&visitor_);
+    Mock::VerifyAndClearExpectations(&return_address_manager_);
+  }
+
+  {
+    UserSpaceFunctionExitPerfEvent function_exit1{
+        .timestamp = 800,
+        .data =
+            {
+                .pid = kPidTargetNamespace,
+                .tid = kTidTargetNamespace,
+            },
+    };
+
+    EXPECT_CALL(return_address_manager_, ProcessFunctionExit(kTidRootNamespace)).Times(1);
+    orbit_grpc_protos::FunctionCall actual_function_call;
+    EXPECT_CALL(listener_, OnFunctionCall).Times(1).WillOnce(SaveArg<0>(&actual_function_call));
+    PerfEvent{function_exit1}.Accept(&visitor_);
+    Mock::VerifyAndClearExpectations(&return_address_manager_);
+    Mock::VerifyAndClearExpectations(&listener_);
+    EXPECT_EQ(actual_function_call.pid(), kPidRootNamespace);
+    EXPECT_EQ(actual_function_call.tid(), kTidRootNamespace);
+    EXPECT_EQ(actual_function_call.function_id(), 1);
+    EXPECT_EQ(actual_function_call.duration_ns(), 500);
+    EXPECT_EQ(actual_function_call.end_timestamp_ns(), 800);
+    EXPECT_EQ(actual_function_call.depth(), 0);
+    EXPECT_EQ(actual_function_call.return_value(), 0);
+    EXPECT_THAT(actual_function_call.registers(), ElementsAre());
+  }
+
+  {
+    UserSpaceFunctionEntryPerfEvent function_entry2{
+        .timestamp = 300,
```

Comment:
Please use increasing timestamps. While it doesn't matter in many of the simple situations in the tests, ordered timestamps is what all our visitors expect.

Reply:
done